src/libraries/System.Private.CoreLib/src/System/Threading/ThreadInt64PersistentCounter.cs

```diff
@@ -9,16 +9,22 @@ namespace System.Threading
 {
     internal sealed class ThreadInt64PersistentCounter
     {
-        private static readonly LowLevelLock s_lock = new LowLevelLock();
+        private readonly LowLevelLock _lock = new LowLevelLock();
 
         [ThreadStatic]
         private static List<ThreadLocalNodeFinalizationHelper>? t_nodeFinalizationHelpers;
 
         private long _overflowCount;
 
-        // we pass comparer explicitly so that in NativeAOT case we would not end up
-        // calling into the type system when lock contention increments its counter
-        private HashSet<ThreadLocalNode> _nodes = new HashSet<ThreadLocalNode>(ObjectEqualityComparer<object>.Default);
+        // dummy node serving as a start and end of the ring list
+        private ThreadLocalNode _first;
```

Comment:
```suggestion
        private ThreadLocalNode _nodes;
```
It is both start and end of the list now, so `_first` is not a fitting name anymore.